<?xml version="1.0"?>
<robot name="centauro">

   <group name="velodyne">
      <chain base_link="torso_2" tip_link="neck_3"/>
   </group>

   <group name="neck">
      <chain base_link="torso_2" tip_link="neck_2"/>
   </group>

   <group name="front_left_leg">
      <chain base_link="pelvis" tip_link="wheel_1"/>
   </group>

   <group name="front_right_leg">
      <chain base_link="pelvis" tip_link="wheel_2"/>
   </group>

   <group name="rear_right_leg">
      <chain base_link="pelvis" tip_link="wheel_4"/>
   </group>

   <group name="rear_left_leg">
      <chain base_link="pelvis" tip_link="wheel_3"/>
   </group>

    <group name="left_arm">
        <chain base_link="torso_2" tip_link="arm1_8"/>
    </group>

    <group name="right_arm">
        <chain base_link="torso_2" tip_link="arm2_8"/>
    </group>

    <group name="torso">
        <chain base_link="pelvis" tip_link="torso_2"/>
    </group>

    <group name="arms">
	<group name="left_arm"/>
        <group name="right_arm"/>
    </group>

    <group name="hands">
      <joint name="j_arm1_8"/>
      <joint name="j_arm2_8"/>
    </group>


   <group name="legs">
      <group name="front_left_leg"/>
      <group name="front_right_leg"/>
      <group name="rear_right_leg"/>
      <group name="rear_left_leg"/>
   </group>


    <group name="force_torque_sensors">
      <joint name="j_ft_1"/>
      <joint name="j_ft_2"/>
    </group>

    <group name="imu_sensors">
       <link name="imu_link"/>
    </group>

   <group name="chains">

    <group name="velodyne"/>
    <group name="neck"/>
    <group name="left_arm"/>
    <group name="right_arm"/>
    <group name="torso"/>
    <group name="front_left_leg"/>
    <group name="front_right_leg"/>
    <group name="rear_right_leg"/>
    <group name="rear_left_leg"/>

   </group>


   <group_state name="home" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="0.5"/>
        <joint name="j_arm1_2" value="-0.3"/>
        <joint name="j_arm1_3" value="-0.3"/>
        <joint name="j_arm1_4" value="-2.2"/>
        <joint name="j_arm1_5" value="0.0"/>
        <joint name="j_arm1_6" value="-0.8"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="-0.5"/>
        <joint name="j_arm2_2" value="0.3"/>
        <joint name="j_arm2_3" value="0.3"/>
        <joint name="j_arm2_4" value="2.2"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="0.8"/>
        <joint name="j_arm2_7" value="0.0"/>
        
        <joint name="hip_yaw_1" value="1.5"/>
        <joint name="hip_pitch_1" value="0.8"/>
        <joint name="knee_pitch_1" value="0.8"/>
        <joint name="ankle_yaw_1" value="1.5"/>

        <joint name="hip_yaw_2" value="-1.5"/>
        <joint name="hip_pitch_2" value="-0.8"/>
        <joint name="knee_pitch_2" value="-0.8"/>
        <joint name="ankle_yaw_2" value="-1.5"/>

        <joint name="hip_yaw_3" value="-1.5"/>
        <joint name="hip_pitch_3" value="-0.8"/>
        <joint name="knee_pitch_3" value="-0.8"/>
        <joint name="ankle_yaw_3" value="-1.5"/>

        <joint name="hip_yaw_4" value="1.5"/>
        <joint name="hip_pitch_4" value="0.8"/>
        <joint name="knee_pitch_4" value="0.8"/>
        <joint name="ankle_yaw_4" value="1.5"/>

<!--        <joint name="hip_yaw_1" value="0.0"/>
        <joint name="hip_pitch_1" value="-1.0"/>
        <joint name="knee_pitch_1" value="-1.0"/>
        <joint name="ankle_yaw_1" value="-0.78"/>

        <joint name="hip_yaw_2" value="0.0"/>
        <joint name="hip_pitch_2" value="1.0"/>
        <joint name="knee_pitch_2" value="1.0"/>
        <joint name="ankle_yaw_2" value="0.78"/>

        <joint name="hip_yaw_3" value="0.0"/>
        <joint name="hip_pitch_3" value="1.0"/>
        <joint name="knee_pitch_3" value="1.0"/>
        <joint name="ankle_yaw_3" value="-0.78"/>

        <joint name="hip_yaw_4" value="0.0"/>
        <joint name="hip_pitch_4" value="-1.0"/>
        <joint name="knee_pitch_4" value="-1.0"/>
        <joint name="ankle_yaw_4" value="0.78"/>-->

    </group_state>


    <!--<group_state name="home" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="0.0"/>
        <joint name="j_arm1_2" value="-0.3"/>
        <joint name="j_arm1_3" value="-0.8"/>
        <joint name="j_arm1_4" value="-0.8"/>
        <joint name="j_arm1_5" value="-0.0"/>
        <joint name="j_arm1_6" value="-0.8"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="0.0"/>
        <joint name="j_arm2_2" value="0.3"/>
        <joint name="j_arm2_3" value="0.8"/>
        <joint name="j_arm2_4" value="0.8"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="0.8"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="1.5"/>
        <joint name="hip_pitch_1" value="0.8"/>
        <joint name="knee_pitch_1" value="0.8"/>
        <joint name="ankle_yaw_1" value="1.5"/>

        <joint name="hip_yaw_2" value="-1.5"/>
        <joint name="hip_pitch_2" value="-0.8"/>
        <joint name="knee_pitch_2" value="-0.8"/>
        <joint name="ankle_yaw_2" value="-1.5"/>

        <joint name="hip_yaw_3" value="-1.5"/>
        <joint name="hip_pitch_3" value="-0.8"/>
        <joint name="knee_pitch_3" value="-0.8"/>
        <joint name="ankle_yaw_3" value="-1.5"/>

        <joint name="hip_yaw_4" value="1.5"/>
        <joint name="hip_pitch_4" value="0.8"/>
        <joint name="knee_pitch_4" value="0.8"/>
        <joint name="ankle_yaw_4" value="1.5"/>

    </group_state>-->

    <group_state name="home_ank_yaw_0" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="0.0"/>
        <joint name="j_arm1_2" value="-0.3"/>
        <joint name="j_arm1_3" value="-0.8"/>
        <joint name="j_arm1_4" value="-0.8"/>
        <joint name="j_arm1_5" value="-0.0"/>
        <joint name="j_arm1_6" value="-0.8"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="0.0"/>
        <joint name="j_arm2_2" value="0.3"/>
        <joint name="j_arm2_3" value="0.8"/>
        <joint name="j_arm2_4" value="0.8"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="0.8"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="1.5"/>
        <joint name="hip_pitch_1" value="0.8"/>
        <joint name="knee_pitch_1" value="0.8"/>
        <joint name="ankle_yaw_1" value="0.0"/>

        <joint name="hip_yaw_2" value="-1.5"/>
        <joint name="hip_pitch_2" value="-0.8"/>
        <joint name="knee_pitch_2" value="-0.8"/>
        <joint name="ankle_yaw_2" value="0.0"/>

        <joint name="hip_yaw_3" value="-1.5"/>
        <joint name="hip_pitch_3" value="-0.8"/>
        <joint name="knee_pitch_3" value="-0.8"/>
        <joint name="ankle_yaw_3" value="0.0"/>

        <joint name="hip_yaw_4" value="1.5"/>
        <joint name="hip_pitch_4" value="0.8"/>
        <joint name="knee_pitch_4" value="0.8"/>
        <joint name="ankle_yaw_4" value="0.0"/>

    </group_state>

    <group_state name="Zero_Pose" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="1.0"/>
        <joint name="j_arm1_2" value="-1.0"/>
        <joint name="j_arm1_3" value="0.0"/>
        <joint name="j_arm1_4" value="0.0"/>
        <joint name="j_arm1_5" value="0.0"/>
        <joint name="j_arm1_6" value="0.0"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="-1.0"/>
        <joint name="j_arm2_2" value="1.0"/>
        <joint name="j_arm2_3" value="0.0"/>
        <joint name="j_arm2_4" value="0.0"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="0.0"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="0.0"/>
        <joint name="hip_pitch_1" value="0.0"/>
        <joint name="knee_pitch_1" value="0.0"/>
        <joint name="ankle_pitch_1" value="0.0"/>
        <joint name="ankle_yaw_1" value="0.0"/>

        <joint name="hip_yaw_2" value="0.0"/>
        <joint name="hip_pitch_2" value="0.0"/>
        <joint name="knee_pitch_2" value="0.0"/>
        <joint name="ankle_pitch_2" value="0.0"/>
        <joint name="ankle_yaw_2" value="0.0"/>

        <joint name="hip_yaw_3" value="0.0"/>
        <joint name="hip_pitch_3" value="0.0"/>
        <joint name="knee_pitch_3" value="0.0"/>
        <joint name="ankle_pitch_3" value="0.0"/>
        <joint name="ankle_yaw_3" value="0.0"/>

        <joint name="hip_yaw_4" value="0.0"/>
        <joint name="hip_pitch_4" value="0.0"/>
        <joint name="knee_pitch_4" value="0.0"/>
        <joint name="ankle_pitch_4" value="0.0"/>
        <joint name="ankle_yaw_4" value="0.0"/>

    </group_state>

    <group_state name="Navvab_Parking_1" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="1.57"/>
        <joint name="j_arm1_2" value="0.1"/>
        <joint name="j_arm1_3" value="1.75"/>
        <joint name="j_arm1_4" value="-1.57"/>
        <joint name="j_arm1_5" value="0.0"/>
        <joint name="j_arm1_6" value="1.57"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="-1.57"/>
        <joint name="j_arm2_2" value="0.1"/>
        <joint name="j_arm2_3" value="-1.75"/>
        <joint name="j_arm2_4" value="1.57"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="-1.57"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="0.0"/>
        <joint name="hip_pitch_1" value="-1.57"/>
        <joint name="knee_pitch_1" value="2.44"/>
        <joint name="ankle_pitch_1" value="2.44"/>
        <joint name="ankle_yaw_1" value="0.0"/>

        <joint name="hip_yaw_2" value="0.0"/>
        <joint name="hip_pitch_2" value="1.57"/>
        <joint name="knee_pitch_2" value="-2.44"/>
        <joint name="ankle_pitch_2" value="-2.44"/>
        <joint name="ankle_yaw_2" value="0.0"/>

        <joint name="hip_yaw_3" value="0.0"/>
        <joint name="hip_pitch_3" value="1.57"/>
        <joint name="knee_pitch_3" value="-2.44"/>
        <joint name="ankle_pitch_3" value="-2.44"/>
        <joint name="ankle_yaw_3" value="0.0"/>

        <joint name="hip_yaw_4" value="0.0"/>
        <joint name="hip_pitch_4" value="-1.57"/>
        <joint name="knee_pitch_4" value="2.44"/>
        <joint name="ankle_pitch_4" value="2.44"/>
        <joint name="ankle_yaw_4" value="0.0"/>

    </group_state>

    <group_state name="Navvab_Parking_2" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="1.57"/>
        <joint name="j_arm1_2" value="0.1"/>
        <joint name="j_arm1_3" value="1.75"/>
        <joint name="j_arm1_4" value="-1.57"/>
        <joint name="j_arm1_5" value="0.0"/>
        <joint name="j_arm1_6" value="1.57"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="-1.57"/>
        <joint name="j_arm2_2" value="0.1"/>
        <joint name="j_arm2_3" value="-1.75"/>
        <joint name="j_arm2_4" value="1.57"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="-1.57"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="0.0"/>
        <joint name="hip_pitch_1" value="-1.83"/>
        <joint name="knee_pitch_1" value="-2.44"/>
        <joint name="ankle_pitch_1" value="0.7"/>
        <joint name="ankle_yaw_1" value="0.0"/>

        <joint name="hip_yaw_2" value="0.0"/>
        <joint name="hip_pitch_2" value="1.83"/>
        <joint name="knee_pitch_2" value="2.44"/>
        <joint name="ankle_pitch_2" value="-0.7"/>
        <joint name="ankle_yaw_2" value="0.0"/>

        <joint name="hip_yaw_3" value="0.0"/>
        <joint name="hip_pitch_3" value="1.83"/>
        <joint name="knee_pitch_3" value="2.44"/>
        <joint name="ankle_pitch_3" value="-0.7"/>
        <joint name="ankle_yaw_3" value="0.0"/>

        <joint name="hip_yaw_4" value="0.0"/>
        <joint name="hip_pitch_4" value="-1.83"/>
        <joint name="knee_pitch_4" value="-2.44"/>
        <joint name="ankle_pitch_4" value="0.7"/>
        <joint name="ankle_yaw_4" value="0.0"/>

    </group_state>

    <group_state name="Navvab_Parking_3" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="1.57"/>
        <joint name="j_arm1_2" value="0.1"/>
        <joint name="j_arm1_3" value="1.75"/>
        <joint name="j_arm1_4" value="-1.57"/>
        <joint name="j_arm1_5" value="0.0"/>
        <joint name="j_arm1_6" value="1.57"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="-1.57"/>
        <joint name="j_arm2_2" value="0.1"/>
        <joint name="j_arm2_3" value="-1.75"/>
        <joint name="j_arm2_4" value="1.57"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="-1.57"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="0.0"/>
        <joint name="hip_pitch_1" value="-1.83"/>
        <joint name="knee_pitch_1" value="-2.44"/>
        <joint name="ankle_pitch_1" value="-2.44"/>
        <joint name="ankle_yaw_1" value="0.0"/>

        <joint name="hip_yaw_2" value="0.0"/>
        <joint name="hip_pitch_2" value="1.83"/>
        <joint name="knee_pitch_2" value="2.44"/>
        <joint name="ankle_pitch_2" value="2.44"/>
        <joint name="ankle_yaw_2" value="0.0"/>

        <joint name="hip_yaw_3" value="0.0"/>
        <joint name="hip_pitch_3" value="1.83"/>
        <joint name="knee_pitch_3" value="2.44"/>
        <joint name="ankle_pitch_3" value="2.44"/>
        <joint name="ankle_yaw_3" value="0.0"/>

        <joint name="hip_yaw_4" value="0.0"/>
        <joint name="hip_pitch_4" value="-1.83"/>
        <joint name="knee_pitch_4" value="-2.44"/>
        <joint name="ankle_pitch_4" value="-2.44"/>
        <joint name="ankle_yaw_4" value="0.0"/>

    </group_state>

    <group_state name="Navvab_Parking_4" group="chains">

        <joint name="torso_yaw" value="0.0"/>

        <joint name="j_arm1_1" value="1.57"/>
        <joint name="j_arm1_2" value="0.1"/>
        <joint name="j_arm1_3" value="1.75"/>
        <joint name="j_arm1_4" value="-1.57"/>
        <joint name="j_arm1_5" value="0.0"/>
        <joint name="j_arm1_6" value="1.57"/>
        <joint name="j_arm1_7" value="0.0"/>

        <joint name="j_arm2_1" value="-1.57"/>
        <joint name="j_arm2_2" value="0.1"/>
        <joint name="j_arm2_3" value="-1.75"/>
        <joint name="j_arm2_4" value="1.57"/>
        <joint name="j_arm2_5" value="0.0"/>
        <joint name="j_arm2_6" value="-1.57"/>
        <joint name="j_arm2_7" value="0.0"/>

        <joint name="hip_yaw_1" value="0.0"/>
        <joint name="hip_pitch_1" value="0.52"/>
        <joint name="knee_pitch_1" value="2.44"/>
        <joint name="ankle_pitch_1" value="2.44"/>
        <joint name="ankle_yaw_1" value="0.0"/>

        <joint name="hip_yaw_2" value="0.0"/>
        <joint name="hip_pitch_2" value="-0.52"/>
        <joint name="knee_pitch_2" value="-2.44"/>
        <joint name="ankle_pitch_2" value="-2.44"/>
        <joint name="ankle_yaw_2" value="0.0"/>

        <joint name="hip_yaw_3" value="0.0"/>
        <joint name="hip_pitch_3" value="-0.52"/>
        <joint name="knee_pitch_3" value="-2.44"/>
        <joint name="ankle_pitch_3" value="-2.44"/>
        <joint name="ankle_yaw_3" value="0.0"/>

        <joint name="hip_yaw_4" value="0.0"/>
        <joint name="hip_pitch_4" value="0.52"/>
        <joint name="knee_pitch_4" value="2.44"/>
        <joint name="ankle_pitch_4" value="2.44"/>
        <joint name="ankle_yaw_4" value="0.0"/>

    </group_state>



</robot>
